### PR TITLE
Implement SafeExchangeCompletionListener utility

### PR DESCRIPTION
**Pull request by carterkozak**
This is a public utility class which can be used elsewhere.

## After this PR
==COMMIT_MSG==
Implement SafeExchangeCompletionListener utility
==COMMIT_MSG==

### CODE REVIEW

File: conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/SafeExchangeCompletionListenerTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.palantir.conjure.java.undertow.HttpServerExchanges;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import io.undertow.server.ExchangeCompletionListener;
+import io.undertow.server.HttpServerExchange;
+import java.util.function.Consumer;
+import org.junit.Test;
```
Comment (ferozco) [inline conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/SafeExchangeCompletionListenerTest.java]: can you merge develop and switch over to junit-jupiter

File: conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/SafeExchangeCompletionListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import org.mockito.Mockito;
+
+public class SafeExchangeCompletionListenerTest {
+
+    private static final HttpServerExchange exchange = HttpServerExchanges.createStub();
+
+    @Test
+    public void test_normalExecution() {
+        ExchangeCompletionListener.NextListener nextListener =
```
Comment (ferozco) [inline conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/SafeExchangeCompletionListenerTest.java]: Can you hoist nextListener and action to be fields with `@Mock` annotation and then add `@ExtendWith(MockitoExtension.class)`

**Comment by ferozco** [comment]
👍 pending cleanup of tests

**Comment by carterkozak** [comment]
Updated, thanks :-)

**Pull request merged**